src/test/java/org/jabref/gui/edit/CopyMoreActionTest.java

```diff
@@ -0,0 +1,168 @@
+    public void testExecuteCopyTitleWithNoTitle() {
+        BibEntry entryWithNoTitle = (BibEntry) entry.clone();
+        entryWithNoTitle.clearField(StandardField.TITLE);
+        ObservableList<BibEntry> entriesWithNoTitles = FXCollections.observableArrayList(entryWithNoTitle);
+        BibDatabaseContext databaseContext = new BibDatabaseContext(new BibDatabase(entriesWithNoTitles));
+
+        when(stateManager.getActiveDatabase()).thenReturn(Optional.ofNullable(databaseContext));
+        when(stateManager.getSelectedEntries()).thenReturn(entriesWithNoTitles);
+        copyMoreAction = new CopyMoreAction(StandardActions.COPY_TITLE, dialogService, stateManager, clipBoardManager, preferencesService);
+        copyMoreAction.execute();
+
+        verify(clipBoardManager, times(0)).setContent(any(String.class));
+        verify(dialogService, times(1)).notify(Localization.lang("None of the selected entries have titles."));
+    }
+
+    @Test
+    public void testExecuteCopyTitleOnPartialSuccess() {
+        BibEntry entryWithNoTitle = (BibEntry) entry.clone();
+        entryWithNoTitle.clearField(StandardField.TITLE);
+        ObservableList<BibEntry> mixedEntries = FXCollections.observableArrayList(entryWithNoTitle, entry);
+        BibDatabaseContext databaseContext = new BibDatabaseContext(new BibDatabase(mixedEntries));
+
+        when(stateManager.getActiveDatabase()).thenReturn(Optional.ofNullable(databaseContext));
+        when(stateManager.getSelectedEntries()).thenReturn(mixedEntries);
+        copyMoreAction = new CopyMoreAction(StandardActions.COPY_TITLE, dialogService, stateManager, clipBoardManager, preferencesService);
+        copyMoreAction.execute();
+
+        String copiedTitles = String.join("\n", titles);
+        verify(clipBoardManager, times(1)).setContent(copiedTitles);
+        verify(dialogService, times(1)).notify(Localization.lang("Warning: %0 out of %1 entries have undefined title.",
+                Integer.toString(mixedEntries.size() - titles.size()), Integer.toString(mixedEntries.size())));
+    }
+
+    @Test
+    public void testExecuteCopyTitleOnSuccess() {
+        ObservableList<BibEntry> entriesWithTitles = FXCollections.observableArrayList(entry);
+        BibDatabaseContext databaseContext = new BibDatabaseContext(new BibDatabase(entriesWithTitles));
+
+        when(stateManager.getActiveDatabase()).thenReturn(Optional.ofNullable(databaseContext));
+        when(stateManager.getSelectedEntries()).thenReturn(entriesWithTitles);
+        copyMoreAction = new CopyMoreAction(StandardActions.COPY_TITLE, dialogService, stateManager, clipBoardManager, preferencesService);
+        copyMoreAction.execute();
+
+        String copiedTitles = String.join("\n", titles);
+        verify(clipBoardManager, times(1)).setContent(copiedTitles);
+        verify(dialogService, times(1)).notify(Localization.lang("Copied '%0' to clipboard.",
+                JabRefDialogService.shortenDialogMessage(copiedTitles)));
```

Comment:
Yes. This setter is no fix for the hack, but a disguise.

Reply:
Oh yes, I just realized this shortenDialogMessage method is being tested as well. However, as I mentioned in a comment above, the reason why I used specific localized argument was because the tests are meant to verify if dialogSerivce is showing the correct message under each condition.  Maybe here I can just simply use any() as the mocked argument since it's too complicated.

Reply:
I think it's okay.